### PR TITLE
Modify driver tests to support swift-driver json and path differences

### PR DESCRIPTION
**Pull request by cltnschlosser**
These changes allow the tests to run (still fail, but with an actual cause): https://github.com/apple/swift-driver/pull/365/commits/4d590a7e91b6b2f6fb38db9bd1b658233bded7d9

### CODE REVIEW

**Review by owenv — APPROVED** [review]
These changes all seem reasonable 👍

**Comment by owenv** [comment]
@swift-ci please test

**Review by DougGregor — APPROVED** [review]
Good idea, thanks!

**Comment by owenv** [comment]
@swift-ci pleas smoke test

**Comment by owenv** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
I had noticed this problem, but not done anything about it (yet). Thanks for addressing it!

File: validation-test/Driver/Dependencies/rdar25405605.swift [hunk 1]
```diff
@@ -28,15 +28,15 @@
...
+// CHECK-1-SKIPPED-DAG: "kind"{{ ?}}: "skipped"
+// CHECK-1-SKIPPED-DAG: "name"{{ ?}}: "compile"
+// CHECK-1-SKIPPED-DAG: "{{(\.\\\/)?}}main.swift"
 // CHECK-1-SKIPPED: {{^}$}}
 
 // CHECK-1-SKIPPED: {{^{$}}
-// CHECK-1-SKIPPED: "kind": "skipped"
-// CHECK-1-SKIPPED: "name": "compile"
-// CHECK-1-SKIPPED: ".\/helper.swift"
+// CHECK-1-SKIPPED-DAG: "kind"{{ ?}} "skipped"
```
Comment (cltnschlosser) [inline validation-test/Driver/Dependencies/rdar25405605.swift:37]:
```suggestion
// CHECK-1-SKIPPED-DAG: "kind"{{ ?}}: "skipped"
```
Add back accidentally removed `:`

**Comment by cltnschlosser** [comment]
@owenv Can you kick off CI again. I ran the tests locally and they passed.

**Comment by owenv** [comment]
@swift-ci please smoke test

**Comment by cltnschlosser** [comment]
@swift-ci Please smoke test Windows platform

**Comment by cltnschlosser** [comment]
@swift-ci please test Windows platform

**Comment by owenv** [comment]
@swift-ci please test Windows platform

**Pull request merged**